io_scene_vrm/importer/vrm_animation_importer.py

```diff
@@ -0,0 +1,737 @@
+    ) -> list["NodeRestPoseTree"]:
+        if not 0 <= node_index < len(node_dicts):
+            return []
+        node_dict = node_dicts[node_index]
+        if not isinstance(node_dict, dict):
+            return []
+
+        translation_float3 = convert.float3_or_none(node_dict.get("translation"))
+        if translation_float3:
+            x, y, z = translation_float3
+            translation = Vector((x, -z, y))
+        else:
+            translation = Vector()
+
+        rotation_float4 = convert.float4_or_none(node_dict.get("rotation"))
+        if rotation_float4:
+            x, y, z, w = rotation_float4
+            rotation = Quaternion((w, x, -z, y)).normalized()
+        else:
+            rotation = Quaternion()
+
+        scale_float3 = convert.float3_or_none(node_dict.get("scale"))
+        if scale_float3:
+            x, y, z = scale_float3
+            scale = Vector((x, -z, y))
+        else:
+            scale = Vector((1, 1, 1))
+
+        local_matrix = (
+            Matrix.Translation(translation)
+            @ rotation.to_matrix().to_4x4()
+            @ Matrix.Diagonal(scale).to_4x4()
+        )
+
+        # TODO: 3要素の場合はオイラー角になるか？
+        # TODO: Matrixだったら分解する
+
+        child_indices = node_dict.get("children")
+        if isinstance(child_indices, list):
+            children = tuple(
+                itertools.chain(
+                    *(
+                        NodeRestPoseTree.build(node_dicts, child_index)
+                        for child_index in child_indices
+                        if isinstance(child_index, int)
+                    )
+                )
+            )
+        else:
+            children = ()
+
+        return [
+            NodeRestPoseTree(
+                node_index=node_index,
+                local_matrix=local_matrix,
+                children=children,
+            )
+        ]
+
+
+class VrmAnimationImporter:
+    @staticmethod
+    def execute(
+        context: bpy.types.Context, path: Path, armature: bpy.types.Object
+    ) -> set[str]:
+        return work_in_progress(context, path, armature)
+
+
+def find_root_node_index(
+    node_dicts: list[Json], node_index: int, skip_node_indices: set[int]
+) -> int:
+    for parent_node_index, node_dict in enumerate(node_dicts):
+        if parent_node_index in skip_node_indices:
+            continue
+        if not isinstance(node_dict, dict):
+            continue
+        child_node_indices = node_dict.get("children")
+        if not isinstance(child_node_indices, list):
+            continue
+        for child_node_index in child_node_indices:
+            if child_node_index in skip_node_indices:
+                continue
+            if child_node_index != node_index:
+                continue
+            skip_node_indices.add(node_index)
+            return find_root_node_index(
+                node_dicts, parent_node_index, skip_node_indices
+            )
+    return node_index
+
+
+def read_accessor_as_bytes(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 16. By comparison, 98% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 176-186 into a separate function.